### DISCUSSION TITLE
"Unsatisfied dependency for type XXX and qualifiers [@Default] "  starting in quarkus 3.9.1

### DISCUSSION THREAD

**Discussion by AviSuli** [quarkusio/quarkus#41455] [Community]
If have an applicationScoped class XXX in my infra jar.
Have also an application, which has this jar as a dependency injected that class.

This injection works fine when I worked with quarkus version 3.8.1.
Once I've upgraded to  3.9.1 I'm getting error:
Unsatisfied dependency for type XXX and qualifiers [@Default] 

I tried to find anything in release notes for 3.9.1 that is related to that, but without success.

Anyone can assist?

Thanks

To give some code snippets:

1. In the jar I have:
```
@ApplicationScoped
public class EsbInfraRouteRequestImpl extends BaseEsbRoute
```

2. In the app
I have dependency on this jar in my pom.xml and in one of the classes I have:
```
@ApplicationScoped
public class RouteRequestAmq2IbmMq extends RouteBuilder{
@Inject
    EsbInfraRouteRequestImpl esbInfraRouteRequest;
```

3. Getting this error:
Unsatisfied dependency for type com.fin.esb.infra.routes.EsbInfraRouteResponseImpl and qualifiers [@Default]
        - injection target: com.fin.esb.routes.RouteRequestAmq2IbmMq#esbInfraRouteResponse
        - declared on CLASS bean [types=[org.apache.camel.builder.BuilderSupport, org.apache.camel.Ordered, org.apache.camel.CamelContextAware, org.apache.camel.spi.ResourceAware, com.fin.esb.routes.RouteRequestAmq2IbmMq, org.a
pache.camel.spi.HasCamelContext, org.apache.camel.RoutesBuilder, java.lang.Object, org.apache.camel.builder.ModelRoutesBuilder, org.apache.camel.builder.RouteBuilder], qualifiers=[@Default, @Any], target=com.fin.esb.routes.RouteRequestAmq2IbmMq]
        at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1516)
        at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:320)

I'll mention again:
When I have in my pom.xml:
`<quarkus.platform.version>3.8.3</quarkus.platform.version>`
All is working fine.
Once I upgraded to `<quarkus.platform.version>3.9.1</quarkus.platform.version>`
I started getting above error.

**Comment by polohero** [comment]
Did you find anything with this? We're getting the same things when upgrading to any version 3.9 and above. It's not clear what the issue is.

**Reply by cescoffier** [reply]
Sorry, I totally missed this issue. Do we know is it's a Camel issue (CC @ppalaga ) or an Arc issue (@mkouba ) ?

**Reply by polohero** [reply]
I have no idea what the issue is to be honest. As far as I can tell it should be seeing the dependency. It builds fine in the 3.8.x, but anything 3.9 and above we get unsatisfied dependency.

I didn't see anything in the migration docs that would explain the difference.

**Reply by mkouba** [reply]
@polohero Do you use Maven or Gradle? Does the dependency have the `beans.xml` descriptor or a Jandex index? It would be great if you could simplify your app to a reproducer that can be shared...
